ql/src/java/org/apache/hadoop/hive/ql/ddl/view/materialized/alter/rewrite/AlterMaterializedViewRewriteOperation.java

```diff
@@ -64,9 +68,12 @@ public int execute() throws HiveException {
           }
           throw new HiveException(msg);
         }
-        if (!planner.isValidAutomaticRewritingMaterialization()) {
-          throw new HiveException("Cannot enable rewriting for materialized view. " +
-              planner.getInvalidAutomaticRewritingMaterializationReason());
+        AutomaticRewritingValidationResult validationResult = planner.getAutomaticRewritingValidationResult();
+        String validationErrorMessage = validationResult.getErrorMessage();
+        if (validationResult.getSupportedRewriteAlgorithms().isEmpty()) {
+          throw new HiveException(validationErrorMessage);
```

Comment:
There seems to be some inconsistency between CREATE and ALTER VIEW statements. Here we throw an exception when there are no rewrite algorithms but in `SemanticAnalyzer#validateCreateView` we simply log an error to the console.

Reply:
Changed that create mv not just log the error but also disables rewrite for MV.